cmake/bob_robotics.cmake

```diff
@@ -207,6 +188,31 @@ macro(BoB_module_custom)
     # Check we're on a supported platform
     check_platform(${PARSED_ARGS_PLATFORMS})
 
+    if(PARSED_ARGS_GAZEBO_PLUGINS)
+        # I'm sometimes getting linker errors when ld is linking against the
+        # static libs for BoB modules (because Gazebo plugins, as shared libs,
+        # are PIC, but the static libs seem not to be). So let's just compile
+        # everything as PIC.
+        if(GNU_TYPE_COMPILER)
+            add_definitions(-fPIC)
+        endif()
+
+        # We need to link against Gazebo libs
+        BoB_external_libraries(gazebo)
+
+        # Dump plugins into source dir
+        set(CMAKE_LIBRARY_OUTPUT_DIRECTORY ${CMAKE_SOURCE_DIR})
+
+        foreach(plugin IN LISTS PARSED_ARGS_GAZEBO_PLUGINS)
+            get_filename_component(shortname ${plugin} NAME)
```

Comment:
what does this do?

Reply:
Gets the filename from the path. Then we trim off the extension to get the target name.

I've stuck in a comment explaining this